src/utils/drag-and-drop-utils.js

```diff
@@ -68,6 +68,9 @@ function canDrop(dropTargetProps, monitor, isHover = false) {
 
     const targetDepth = getTargetDepth(dropTargetProps, monitor);
     const draggedNode = monitor.getItem().node;
+    const parentPath = dropTargetProps.path.slice(0, -1);
```

Comment:
Well this is not a problem anymore for this PR since `canDrop` now always returns true while dragging, even if `maxDepth` or `canHaveChildren` would prevent drop. That way we can display the red box, which is a better UX